homeassistant/components/contact_energy_nz/sensor.py

```diff
@@ -0,0 +1,222 @@
+"""Support for retrieving usage data from Contact Energy."""
+
+from collections.abc import Callable, Mapping
+from dataclasses import dataclass
+from datetime import datetime, timedelta
+import logging
+from typing import Any, cast
+
+import async_timeout
+from contact_energy_nz import ContactEnergyApi, UsageDatum
+
+from homeassistant.components.sensor import (
+    DOMAIN as SENSOR_DOMAIN,
+    SensorDeviceClass,
+    SensorEntity,
+    SensorEntityDescription,
+    SensorStateClass,
+)
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.const import UnitOfEnergy
+from homeassistant.core import HomeAssistant
+from homeassistant.exceptions import InvalidStateError
+from homeassistant.helpers.device_registry import DeviceEntryType
+from homeassistant.helpers.entity import DeviceInfo
+from homeassistant.helpers.entity_platform import AddEntitiesCallback
+from homeassistant.helpers.update_coordinator import (
+    CoordinatorEntity,
+    DataUpdateCoordinator,
+    UpdateFailed,
+)
+
+from .const import DOMAIN, UPDATE_INTERVAL_HOURS
+
+ENTITY_ID_SENSOR_FORMAT = SENSOR_DOMAIN + ".contact_enegry_nz_{}"
+
+
+@dataclass
+class ContactEnergyUsageMixin:
+    """Extra fields to pass."""
+
+    value_fn: Callable[[UsageDatum], float]
+    unit_fn: Callable[[UsageDatum], str]
+
+
+@dataclass
+class ContactEnergyUsageSensorEntityDescription(
+    SensorEntityDescription, ContactEnergyUsageMixin
+):
+    """Describes sensor entity."""
+
+
+_LOGGER = logging.getLogger(DOMAIN)
+
+SENSOR_TYPES: tuple[ContactEnergyUsageSensorEntityDescription, ...] = (
```

Comment:
New integrations/entities must support [translation of names and attributes](https://developers.home-assistant.io/blog/2023/03/27/entity_name_translations/).

Please adopt your code accordingly.